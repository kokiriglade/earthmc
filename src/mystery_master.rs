//! # Mystery Master
//!
//! Defines the [`MysteryMaster`] struct.
use serde::{Deserialize, Serialize};
use uuid::Uuid;

/// Represents a player participating in "Mystery Master". Change indiciates in
/// what direction they have moved on the scoreboard.
#[derive(Serialize, Deserialize, Debug, PartialEq)]
pub struct MysteryMaster {
    /// The player's username.
    pub name: String,
    /// The player's [`Uuid`].
    pub uuid: Uuid,
    /// The change kind.
    pub change: MysteryMasterChangeKind,
}

#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum MysteryMasterChangeKind {
    Down,
    Unchanged,
    Up,
}
