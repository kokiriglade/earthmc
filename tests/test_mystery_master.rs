use earthmc::mystery_master::MysteryMaster;

#[test]
fn test_mystery_master() {
    let raw_json = include_str!("inputs/mystery_master.json");
    let parsed: Vec<MysteryMaster> = serde_json::from_str(raw_json).unwrap();
    insta::assert_debug_snapshot!(parsed);
}
