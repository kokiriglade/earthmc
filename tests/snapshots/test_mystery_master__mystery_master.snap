---
source: tests/test_mystery_master.rs
expression: parsed
---
[
    MysteryMaster {
        name: "Xin_hachamecha",
        uuid: 8b543488-dd1d-4440-aa44-457fa25249fc,
        change: Up,
    },
    MysteryMaster {
        name: "kazucc",
        uuid: 21dc99b4-b0dc-4aa5-9913-3b5ff2594229,
        change: Up,
    },
    MysteryMaster {
        name: "UrbanTracker_lol",
        uuid: 6743993e-bdd2-4835-90f3-2e8cd14dccf7,
        change: Down,
    },
    MysteryMaster {
        name: "Nutsuko",
        uuid: 4e6f74b3-b260-447e-98bf-a025b728eb4b,
        change: Up,
    },
    MysteryMaster {
        name: "psiroq",
        uuid: 2673751e-5866-42a7-92ea-24bacd856c73,
        change: Down,
    },
    MysteryMaster {
        name: "junjun4654",
        uuid: e3c6db0a-2f6b-4430-988c-610a9ccd659b,
        change: Up,
    },
    MysteryMaster {
        name: "seychelles4567",
        uuid: c445b40f-bce5-4d23-98b1-b31b92f16ae7,
        change: Up,
    },
    MysteryMaster {
        name: "Pamena",
        uuid: 554f4bb7-d663-4c1b-a5b3-c70758e63792,
        change: Down,
    },
    MysteryMaster {
        name: "Vibezisacutie",
        uuid: 96dcee81-ef04-4104-83bc-8e151b3bf58c,
        change: Up,
    },
    MysteryMaster {
        name: "Ariohc",
        uuid: 9c28d713-79e7-41f4-aaed-5b4ab81da015,
        change: Unchanged,
    },
]
